src/utils/getPreviousValue.ts

```diff
@@ -1,6 +1,6 @@
 import { useEffect, useRef } from 'react';
 
-export default function usePreviousValue(value: any): any {
+export default function usePreviousValue<T>(value: T): T {
   const ref = useRef(null);
```

Comment:
There's only one more thing lacking now and that is that null can't be of your generic, this means that you should change this to for example `undefined as T` since it does not have any effect on the eventual code